### PR TITLE
 Add all owrequest adapter logic from helix-cli #46

### PR DESCRIPTION
**Pull request by tripodsan**
see #46 

- provides new openwhisk adapter functionality in `openwhisk.js`
- moves adaptOWRequest and adaptOWResponse out of pipeline

### CODE REVIEW

**Comment by trieloff** [comment]
This pull request **introduces 1 alert** and **fixes 1** when merging 891cca66b9df9a31d49f2e816cd8f43f0467fd62 into 0e3e51c4ea7b1b28180d0223e68a57fd684f7d2a - [view on LGTM.com](https://lgtm.com/projects/g/adobe/hypermedia-pipeline/rev/pr-f1b9304af92858794e12325440f9f941a9d07f5e)

**new alerts:**

* 1 for Property access on null or undefined

**fixed alerts:**

* 1 for Property access on null or undefined

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/utils/openwhisk.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+function extractClientRequest(action) {
+  const { request } = action;
+  try { // the edge encodes the client request parameters into the `params` param ;-)
+    if (!request || !request.params) {
+      return {};
+    }
+    if (request.params.params) {
+      return {
+        params: querystring.parse(request.params.params),
+        headers: request.headers,
```
Comment (trieloff) [inline src/utils/openwhisk.js]: Wouldn't this be in `__ow_headers`?

File: src/utils/openwhisk.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        method: request.method,
```
Comment (trieloff) [inline src/utils/openwhisk.js]: And this in `__ow_method`?

File: src/utils/openwhisk.js (hunk 1 above)
Comment (trieloff) [reply src/utils/openwhisk.js]: I see that `extractActionContext ` is run beforehand now and changes `__ow_headers` to just `headers`. You might want to clarify that in the JSDoc, as the `action` object is not entirely-OW-style.

Comment (tripodsan) [reply src/utils/openwhisk.js]: it is already:

```
 * @param {Object} action The pipeline action context
```

**Comment by trieloff** [comment]
This pull request **introduces 1 alert** and **fixes 1** when merging 926e8ef9ddcf0bb56e0b5e064f8c531a0b82fdf1 into c3eb45ecbbda040befbf1b92590fdc56bcd531a1 - [view on LGTM.com](https://lgtm.com/projects/g/adobe/hypermedia-pipeline/rev/pr-8077b2ec0e09e15d553ea684c6d1a94838efbd8b)

**new alerts:**

* 1 for Property access on null or undefined

**fixed alerts:**

* 1 for Property access on null or undefined

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by trieloff** [comment]
This pull request **introduces 1 alert** and **fixes 1** when merging 1b5842bcde17f73184862139ed3173a4cde52424 into c3eb45ecbbda040befbf1b92590fdc56bcd531a1 - [view on LGTM.com](https://lgtm.com/projects/g/adobe/hypermedia-pipeline/rev/pr-89120dfd97183cecb19e93f0d448d12a6b68d80d)

**new alerts:**

* 1 for Property access on null or undefined

**fixed alerts:**

* 1 for Property access on null or undefined

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**